manuals/1.0/en/database.md

```diff
@@ -180,6 +212,349 @@ use Ray\AuraSqlModule\Annotation\Transactional;
+                'sub.id = foo.id'           // ON these conditions
+            )
+            ->where('bar > :bar')           // AND WHERE these conditions
+            ->where('zim = ?', 'zim_val')   // bind 'zim_val' to the ? placeholder
+            ->orWhere('baz < :baz')         // OR WHERE these conditions
+            ->groupBy(['dib'])              // GROUP BY these columns
+            ->having('foo = :foo')          // AND HAVING these conditions
+            ->having('bar > ?', 'bar_val')  // bind 'bar_val' to the ? placeholder
+            ->orHaving('baz < :baz')        // OR HAVING these conditions
+            ->orderBy(['baz'])              // ORDER BY these columns
+            ->limit(10)                     // LIMIT 10
+            ->offset(40)                    // OFFSET 40
+            ->forUpdate()                   // FOR UPDATE
+            ->union()                       // UNION with a followup SELECT
+            ->unionAll()                    // UNION ALL with a followup SELECT
+            ->bindValue('foo', 'foo_val')   // bind one value to a placeholder
+            ->bindValues([                  // bind these values to named placeholders
+                'bar' => 'bar_val',
+                'baz' => 'baz_val',
+            ]);
+
+        $sth = $this->pdo->prepare($this->select->getStatement());
+
+        // bind the values and execute
+        $sth->execute($this->select->getBindValues());
+        $result = $sth->fetch(\PDO::FETCH_ASSOC);
+        // or
+        // $result = $this->pdo->fetchAssoc($stm, $bind);
+```
+
+The created queries are queried as strings with the `getStatement()`.
+
+## INSERT
+
+### Single row INSERT
+
+
+```php?start_inline
+use Ray\AuraSqlModule\AuraSqlInject;
+use Ray\AuraSqlModule\AuraSqlInsertInject;
+
+class User extend ResourceObject
+{
+    use AuraSqlInject;
+    use AuraSqlInsertInject;
+
+    public function onPost()
+    {
+        $this->insert
+            ->into('foo')                   // INTO this table
+            ->cols([                        // bind values as "(col) VALUES (:col)"
+                'bar',
+                'baz',
+            ])
+            ->set('ts', 'NOW()')            // raw value as "(ts) VALUES (NOW())"
+            ->bindValue('foo', 'foo_val')   // bind one value to a placeholder
+            ->bindValues([                  // bind these values
+                'bar' => 'foo',
+                'baz' => 'zim',
+            ]);
+
+        $sth = $this->pdo->prepare($this->insert->getStatement());
+        $sth->execute($this->insert->getBindValues());
+        // or
+        // $sth = $this->pdo->perform($this->insert->getStatement(), this->insert->getBindValues());
+
+        // get the last insert ID
+        $name = $insert->getLastInsertIdName('id');
+        $id = $pdo->lastInsertId($name);
+```
+
+The `cols()` method can also pass an associative array with the column name as a key and the value is a bind value.
```

Comment:
I got it (from their site):

The cols() method allows you to pass an array of key-value pairs where the key is the column name and the value is a bind value (not a raw value)